src/theory/ext_theory.cpp

```diff
@@ -216,122 +203,90 @@ void ExtTheory::getSubstitutedTerms(int effort,
+        Trace("extt-debug") << "...send lemma " << lem << std::endl;
+        if (sendLemma(trn, InferenceId::EXTT_SIMPLIFY))
+        {
+          Trace("extt-lemma")
+              << "ExtTheory : substitution + rewrite lemma : " << lem
+              << std::endl;
+          addedLemma = true;
         }
-        markInactive(n, ExtReducedId::REDUCTION, satDep);
       }
-    }
-  }
-  else
-  {
-    std::vector<Node> sterms;
-    std::vector<std::vector<Node> > exp;
-    getSubstitutedTerms(effort, terms, sterms, exp);
-    std::map<Node, unsigned> sterm_index;
-    NodeManager* nm = nodeManager();
-    for (unsigned i = 0, size = terms.size(); i < size; i++)
-    {
-      bool processed = false;
-      if (sterms[i] != terms[i])
+      else
       {
-        Node sr = rewrite(sterms[i]);
-        // ask the theory if this term is reduced, e.g. is it constant or it
-        // is a non-extf term.
-        ExtReducedId id;
-        if (d_parent.isExtfReduced(effort, sr, terms[i], exp[i], id))
+        // check if we have already reduced this
+        std::map<Node, unsigned>::iterator itsi = sterm_index.find(sr);
+        if (itsi == sterm_index.end())
         {
-          processed = true;
-          markInactive(terms[i], id);
-          // We have exp[i] => terms[i] = sr, convert this to a clause.
-          // This ensures the proof infrastructure can process this as a
-          // normal theory lemma.
-          Node eq = terms[i].eqNode(sr);
-          Node lem = eq;
-          if (!exp[i].empty())
-          {
-            std::vector<Node> eei;
-            for (const Node& e : exp[i])
-            {
-              eei.push_back(e.negate());
-            }
-            eei.push_back(eq);
-            lem = nm->mkNode(Kind::OR, eei);
-          }
-
-          Trace("extt-debug") << "ExtTheory::doInferences : infer : " << eq
-                              << " by " << exp[i] << std::endl;
-          Trace("extt-debug") << "...send lemma " << lem << std::endl;
-          if (sendLemma(lem, InferenceId::EXTT_SIMPLIFY))
-          {
-            Trace("extt-lemma")
-                << "ExtTheory : substitution + rewrite lemma : " << lem
-                << std::endl;
-            addedLemma = true;
-          }
+          sterm_index[sr] = i;
         }
         else
         {
-          // check if we have already reduced this
-          std::map<Node, unsigned>::iterator itsi = sterm_index.find(sr);
-          if (itsi == sterm_index.end())
-          {
-            sterm_index[sr] = i;
-          }
-          else
-          {
-            // unsigned j = itsi->second;
-            // note : can add (non-reducing) lemma :
-            //   exp[j] ^ exp[i] => sterms[i] = sterms[j]
-          }
-
-          Trace("extt-nred") << "Non-reduced term : " << sr << std::endl;
+          // unsigned j = itsi->second;
+          // note : can add (non-reducing) lemma :
+          //   exp[j] ^ exp[i] => sterms[i] = sterms[j]
```

Comment:
delete the unused code, add a wishue for this potential improvement

Reply:
I don't think this warrants a wishue.

Btw note that all of these comments are for old code that changed indendentation.

Reply:
I've refactored the comment and deleted a few things.